Solve issue 527 -  update scripts to use @salesforce/cli (SF cli)

JMercie opened the pull request:
As mentioned in issue #527, I updated existing scripts in the repository to use SF CLI instead of SFDX CLI to move to the recommended command line tool for Salesforce development, which is the one getting maintained currently. 

Each script was replaced with its SF cli counterpart as detailed [in the documentation](https://developer.salesforce.com/docs/atlas.en-us.sfdx_cli_reference.meta/sfdx_cli_reference/cli_reference_top.htm)

JMercie commented:
@jongpie This PR is ready for review whenever you find time to. Thanks in advance for the time and for any recommendation

jongpie commented on scripts/build/create-scratch-org.ps1:
```diff
@@ -29,22 +29,22 @@ Write-Output "Scratch org duration in days: $durationdays"
             }
 
-            $defaultUsername = (npx sfdx config:get defaultusername  --json | ConvertFrom-Json).result.value
+            $defaultUsername = (npx sf config get target-dev-hub  --json | ConvertFrom-Json).result.value
```
This should use `target-org` instead of `target-dev-hub`

```suggestion
            $defaultUsername = (npx sf config get target-org --json | ConvertFrom-Json).result.value
```

jongpie commented on scripts/build/create-scratch-org.ps1:
```diff
@@ -55,7 +55,7 @@ foreach($devHub in $devHubs) {
 }
 
-$defaultUsername = (npx sfdx config:get defaultusername  --json | ConvertFrom-Json).result.value
+$defaultUsername = (npx sf config get target-dev-hub  --json | ConvertFrom-Json).result.value
```
Same here, this should use `target-org` instead of `target-dev-hub`

```suggestion
$defaultUsername = (npx sf config get target-org  --json | ConvertFrom-Json).result.value
```

jongpie commented on scripts/build/get-devhub-org-details.ps1:
```diff
@@ -5,11 +5,11 @@ param ([string]$devHub)
 Write-Output "Getting dev hub limits for dev hub: $devHub"
-npx sfdx force:org:display --json --verbose --targetusername $devHub
\ No newline at end of file
+npx sf org display --json --verbose -o $devHub
```
```suggestion
npx sf org display --json --verbose --target-org $devHub
```

jongpie reviewed (changes requested):
@JMercie thanks for working on this! I read through the changes, and overall, things look great. There were a few items that I have feedback for - I've added comments about possible changes. Please let me know if you have any questions!

JMercie commented:
Hey @jongpie 

Thanks a lot for the feedback on this!

I also modified a few other commands beyond the ones you suggested. I used the abbreviature for the parameters; now, every command should use the long version.

I reverted the changes in the README.md and create-and-install-package-version.ps1 files, added back the sfdx-cli to the Dev dependencies in the package.json and downgrade the SF cli to 1.87.0 version because until we only want to use the SF cli we won't be able to use the v2 of it.

I hope everything looks better, and if you have any further comments on what has to be changed, I will work on it!

jongpie reviewed (approved):
@JMercie thanks so much for incorporating those changes! 🥳 I've re-reviewed the PR, and I think everything looks great to me. I'll merge this to `main` now, and then I'll test the `build.yml` changes in my open PR #535.

JMercie commented:
> @JMercie thanks so much for incorporating those changes! 🥳 I've re-reviewed the PR, and I think everything looks great to me. I'll merge this to `main` now, and then I'll test the `build.yml` changes in my open PR #535.

Thank you very much for letting people help on this project and for your review on the work!

Looking forward to keep contributing on this project
